plugins/ci/pkg/ci/trivy.go

```diff
@@ -214,10 +214,23 @@ func mergeImages(folderPath string, dockerImages []trivymodels.DockerImage, mani
 		// namely the Owner info (i.e. the deployment or other controller it is associated with)
 		var image *trivymodels.Image
 		for _, im := range manifestImages {
+
+			// match by pullRef
 			if im.PullRef == filename {
 				image = &im
 				break
 			}
+
+			// fallback - match by name
+			if imageName, ok := filenameToImageName[filename]; ok {
+				if im.Name == imageName {
+					if im.PullRef == "" {
+						im.PullRef = filename // set pullRef if not already set
+					}
+					image = &im
+					break
+				}
+			}
```

Comment:
🤔 setting `PullRef` to `filename` seems odd--though it looks like we were already doing that above...

Reply:
In all my tests `PullRef` was blank, this is kinda of a reverse-lookup using the filename to imagemap, while preserving the original logic

But yea, kinda odd indeed... I will give it a deeper look.

Reply:
I was mutating a copy of the slice, that's why `PullRef` was always `""` - fixed that, now we don't need this fallback behavior - thanks for noticing it 🕺